src/main/java/uk/gov/hmcts/cmc/claimstore/services/OffersService.java

```diff
@@ -31,53 +33,67 @@ public OffersService(
-        eventProducer.createOfferMadeEvent(claim);
+        caseRepository.updateSettlement(claim, settlement, authorisation, userAction("OFFER_MADE_BY", party.name()));
+        Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
+        eventProducer.createOfferMadeEvent(updated);
+        return updated;
     }
 
-    public void accept(Claim claim, MadeBy party, String authorisation) {
+    @Transactional
+    public Claim accept(Claim claim, MadeBy party, String authorisation) {
         assertSettlementIsNotReached(claim);
 
         Settlement settlement = claim.getSettlement()
-            .orElseThrow(() -> new ConflictException("Offer has not been made yet."));
+            .orElseThrow(conflictOfferIsNotMade());
 
         settlement.accept(party);
 
         caseRepository.updateSettlement(claim, settlement, authorisation,
-            userAcion("OFFER_ACCEPTED_BY", party.name()));
-        eventProducer.createOfferAcceptedEvent(claim, party);
+            userAction("OFFER_ACCEPTED_BY", party.name()));
+
+        Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
+        eventProducer.createOfferAcceptedEvent(updated, party);
+        return updated;
     }
 
-    public void reject(Claim claim, MadeBy party, String authorisation) {
+    @Transactional
+    public Claim reject(Claim claim, MadeBy party, String authorisation) {
         assertSettlementIsNotReached(claim);
 
         Settlement settlement = claim.getSettlement()
-            .orElseThrow(() -> new ConflictException("Offer has not been made yet."));
+            .orElseThrow(conflictOfferIsNotMade());
         settlement.reject(party);
 
-        caseRepository.updateSettlement(claim, settlement, authorisation, userAcion("OFFER_REJECTED_BY", party.name()));
-
-        eventProducer.createOfferRejectedEvent(claim, party);
+        String userAction = userAction("OFFER_REJECTED_BY", party.name());
+        caseRepository.updateSettlement(claim, settlement, authorisation, userAction);
+        Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
+        eventProducer.createOfferRejectedEvent(updated, party);
+        return updated;
     }
 
     @Transactional
-    public void countersign(Claim claim, MadeBy party, String authorisation) {
+    public Claim countersign(Claim claim, MadeBy party, String authorisation) {
         assertSettlementIsNotReached(claim);
 
         Settlement settlement = claim.getSettlement()
-            .orElseThrow(() -> new ConflictException("Offer has not been made yet."));
+            .orElseThrow(conflictOfferIsNotMade());
         settlement.countersign(party);
 
-        caseRepository.reachSettlementAgreement(claim, settlement, authorisation,
-            userAcion("OFFER_COUNTERSIGNED_BY", party.name()));
+        caseRepository.reachSettlementAgreement(claim, settlement, authorisation, "SETTLED_PRE_JUDGMENT");
```

Comment:
Enum constant `.name()` can be used here.

Reply:
used, thanks